prov/util/src/util_coll.c

```diff
@@ -474,13 +475,13 @@ void util_coll_join_comp(struct util_coll_mc *coll_mc,
 			 struct util_coll_comp_item *comp)
 {
 	struct fi_eq_err_entry entry;
-	struct util_coll_join_comp_data *comp_data;
+	struct util_coll_comp_data *comp_data;
 	struct util_ep *ep;
 	ssize_t bytes;
 	uint64_t tmp;
 	int iter, lsb_set_pos = 0, pos;
 
-	comp_data = (struct util_coll_join_comp_data *)comp->data;
+	comp_data = (struct util_coll_comp_data *)comp->data;
```

Comment:
This is changing existing code, but if comp->data is a pointer to a struct, can we just define it as that, rather than a void *?